syntax = "proto3";

service BeeServer {
  rpc StartBee (StartRequest) returns (StartReply) {}
  rpc SubmitWord (SubmitWordRequest) returns(SubmitWordReply) {}
  rpc CreateBee (CreateRequest) returns (CreateReply) {}
}

message StartRequest{
}

message StartReply{
  string message = 1;
}

message SubmitWordRequest{
  string wordIn = 1;
}

message SubmitWordReply{
  int32 result = 1;
}

message CreateRequest{
  int32 beeType = 1;
}

message CreateReply{
  string message = 1;
}

message JoinRequest{
  string gameID = 1;
  string name = 2;
}

message JoinReply{
  string joinMessage = 1;
}
